js-packages/web-console/src/lib/compositions/usePipelineAction.svelte.ts

```diff
@@ -64,8 +69,9 @@ export const usePipelineAction = () => {
     'Replaying'
   ]
   const reactiveWaiter = useReactiveWaiter(() => pipelineList.pipelines)
-  return {
-    postPipelineAction: async (
+  // We initialize a global state that will be reused across all calls to `getPipelineAction`.
+  // This allows keeping the same waiters alive while switching UI context
```

Comment:
I don't know who the waiters are or what switching UI context means, but perhaps I have missed too many reviews of this code to understand